### DISCUSSION TITLE
git-repo external merge with files

### DISCUSSION THREAD

**Discussion by mirsella** [twpayne/chezmoi#3116] [Q&A] [answered]
hello,

i have this .chezmoiexternal 
```
[".config/nvim"]
  type = "git-repo"
  url = "https://github.com/LazyVim/starter"
  refreshPeriod = "168h"
```
but I would like to add custom config which are located in `<lazyvim root, so .config/nvim>/config/plugins/file.lua`.
i tried to add `.config/nvim/config/plugins/file.lua` to chezmoi, but when applying and so cloning the repo it says
```
chezmoi: .config/nvim: inconsistent state (/home/mirsella/.local/share/chezmoi/private_dot_config/nvim, https://github.com/LazyVim/starter defined in /home/mirsella/.local/share/chezmoi/.chezmoiexternal.toml)
```
Which I expected to fail, but I wonder if there is a better way to do it than run_ script to cp the file.

thanks !

**Comment by 2-4601** [comment, marked as answer]
Not sure if I completely understand what you are after but as the [docs say](https://www.chezmoi.io/user-guide/include-files-from-elsewhere/#include-a-subdirectory-from-a-git-repository):

> Using a `git-repo` external delegates management of the directory to git. chezmoi cannot manage any other files in that directory.

I have not used LazyVim but in its [installation docs they tell](https://www.lazyvim.org/installation) you to remove the `.git` folder so maybe it's not meant to be used as a git repository after the initial cloning?

As GitHub also allows downloading repositories as zip packages, you could use `.chezmoiexternal`'s _archive_ type instead of the _git-repo_ type like this:

```ini
[".config/neovim"]
    type = "archive"
    stripComponents = 1
    url = "https://github.com/LazyVim/starter/archive/refs/heads/main.zip"
    refreshPeriod = "168h"
    exclude = ["*/README*", "*/LICENSE", "*/.gitignore"]
```

This should allow you to manage other files in `.config/neovim` with chezmoi.

**Comment by mirsella** [comment]
thank you ! the archive type is what I was looking for.
I initially thought all the external couldn't work with local file
